src/index.js

```diff
@@ -0,0 +1,8 @@
+'use strict'
+
+module.exports = {
+  Stop: require('./circuit/hop'),
+  Dialer: require('./dialer'),
+  multicodec: require('./multicodec'),
+  tag: 'Circuit'
+}
```

Comment:
@dryajov's last comment pretty much covers the intentions of having the relay be a transport. It's the way of least risk, and least edge-cases.

> tying up Relay as transport forces it to follow everything a transport has, which is just going to be a pain in the long term and confusing right now.

I disagree -- which pains do you anticipate? I see how it's a bit mind-bending at the beginning, but the gains are sooo worth it. No special-case code spread accross libp2p, no reimplementation of things like secureConn (not sure how the secio wiring is called in js-libp2p) -- we get it all for free.

Also keep in mind that this particular relay (circuit) will definitely not be the last of its kind.